### ISSUE TITLE
Accuracy for multi-task output binary classification not working properly

### ISSUE THREAD

**Issue by daquang** [fchollet/keras#2412]
It appears that versions 1.0.0 and above are not calculating accuracy for multi-task output binary classification problems correctly. For example, suppose my target for each sample is a 1,000x1 binary vector. In previous versions of keras, the accuracy would be computer correctly correctly. I managed to fix this issue with a crude hack. In the training.py script in engine, I changed (lines 598-604):

            for metric in output_metrics:
                if metric == 'accuracy' or metric == 'acc':
                    # custom handling of accuracy (because of class mode duality)
                    output_shape = self.internal_output_shapes[i]
                    if output_shape[-1] == 1:
                        # case: binary accuracy
                        self.metrics.append(metrics_module.binary_accuracy(y_true, y_pred))

to:

            for metric in output_metrics:
                if metric == 'accuracy' or metric == 'acc':
                    # custom handling of accuracy (because of class mode duality)
                    output_shape = self.internal_output_shapes[i]
                    if output_shape[-1] == 1 or loss == 'binary_crossentropy':
                        # case: binary accuracy
                        self.metrics.append(metrics_module.binary_accuracy(y_true, y_pred))

**Comment by fchollet** [comment]
The official solution to this problem is to use:

`model.compile(..., metrics=['binary_accuracy'])`

The check we are doing to route `acc` to either `categorical_accuracy` or `binary_accuracy` is basic, so it can fail. It's not generally possible to guess what the user meant, so we only do this on a best effort basis. If you don't want Keras to guess for you, be explicit about what you mean.
